x-pack/plugins/reporting/server/export_types/common/pdf/pdfmaker.ts

```diff
@@ -124,37 +142,64 @@ export class PdfMaker {
     this._title = title;
   }
 
-  generate() {
-    const docTemplate = _.assign(
-      getTemplate(this._layout, this._logo, this._title, tableBorderWidth, assetPath),
-      {
-        content: this._content,
-      }
-    );
-    this._pdfDoc = this._printer.createPdfKitDocument(docTemplate, getDocOptions(tableBorderWidth));
-    return this;
+  private getWorkerData(): PdfWorkerData {
+    return {
+      layout: {
+        hasFooter: this._layout.hasFooter,
+        hasHeader: this._layout.hasHeader,
+        orientation: this._layout.getPdfPageOrientation(),
+        useReportingBranding: this._layout.useReportingBranding,
+        pageSize: this._layout.getPdfPageSize({
+          pageMarginTop,
+          pageMarginBottom,
+          pageMarginWidth,
+          tableBorderWidth,
+          headingHeight,
+          subheadingHeight,
+        }),
+      },
+      title: this._title,
+      logo: this._logo,
+      content: this._content as unknown as SerializableRecord[],
+    };
   }
 
-  getBuffer(): Promise<Buffer | null> {
-    return new Promise((resolve, reject) => {
-      if (!this._pdfDoc) {
-        throw new Error(
-          i18n.translate(
-            'xpack.reporting.exportTypes.printablePdf.documentStreamIsNotgeneratedErrorMessage',
-            {
-              defaultMessage: 'Document stream has not been generated',
-            }
-          )
-        );
-      }
-
-      const concatStream = concat(function (pdfBuffer: Buffer) {
-        resolve(pdfBuffer);
+  public async generate(): Promise<Uint8Array> {
+    if (this.worker) throw new Error('PDF generation already in progress!');
```

Comment:
After Monitoring of Reporting, when users have more insight to what is happening inside of Reporting, they should be able to accept the additional load of multiple reports. This is probably OK for now, but at some point (after Monitoring of Reporting), we should support multiple concurrent reports executing / compiling.